src/daft-local-execution/src/run.rs

```diff
@@ -86,9 +104,93 @@ pub fn run_local(
 ) -> DaftResult<Box<dyn Iterator<Item = DaftResult<Arc<MicroPartition>>> + Send>> {
     let runtime = tokio::runtime::Runtime::new().expect("Failed to create tokio runtime");
     let res = runtime.block_on(async {
-        let pipeline = physical_plan_to_pipeline(physical_plan, &psets);
-        let stream = pipeline.get_data().await;
-        stream.collect::<Vec<_>>().await
+        let (sender, mut receiver) = create_channel(1, true);
+        run_physical_plan(physical_plan, &psets, sender);
+        let mut result = vec![];
+        while let Some(val) = receiver.recv().await {
+            result.push(val);
+        }
+        result.into_iter()
     });
-    Ok(Box::new(res.into_iter()))
+    Ok(Box::new(res))
+}
+
+pub fn run_physical_plan(
```

Comment:
decouple, building pipeline and starting execution.

1. build full pipeline
2. run `start` for each op
3. then wait on result